### PR TITLE
Convert covidcast signal OR clauses to UNION

### LINKED ISSUE #763: Convert covidcast signal OR clauses to UNION
The following query returns 248 rows and takes 30 minutes to compute:

```
SELECT t.geo_type, t.geo_value, t.source, t.signal, t.time_value, t.value 
FROM covidcast t 
WHERE ((t.source = 'jhu-csse' 
AND (t.signal = 'confirmed_7dav_incidence_prop' OR t.signal = 'deaths_7dav_incidence_prop'))) 
AND ((t.geo_type = 'nation' AND (t.geo_value = 'us'))) 
AND ((t.time_type = 'day' AND (t.time_value BETWEEN 20210706 AND 20211106))) 
AND (t.is_latest_issue IS TRUE) 
ORDER BY t.geo_type ASC, t.geo_value ASC, t.source ASC, t.signal ASC, t.time_value ASC 
LIMIT 10000001;
```

Either one of the signals on its own runs in less than 1s each, as does the UNION of the two:

```
SELECT t.geo_type, t.geo_value, t.source, t.signal, t.time_value, t.value 
FROM covidcast t 
WHERE ((t.source = 'jhu-csse' AND (t.signal = 'confirmed_7dav_incidence_prop'))) 
AND ((t.geo_type = 'nation' AND (t.geo_value = 'us'))) 
AND ((t.time_type = 'day' AND (t.time_value BETWEEN 20210706 AND 20211106))) 
AND (t.is_latest_issue IS TRUE) 
UNION ALL 
SELECT t.geo_type, t.geo_value, t.source, t.signal, t.time_value, t.value 
FROM covidcast t 
WHERE ((t.source = 'jhu-csse' AND (t.signal = 'deaths_7dav_incidence_prop'))) 
AND ((t.geo_type = 'nation' AND (t.geo_value = 'us'))) 
AND ((t.time_type = 'day' AND (t.time_value BETWEEN 20210706 AND 20211106))) 
AND (t.is_latest_issue IS TRUE) 
ORDER BY geo_type ASC, geo_value ASC, source ASC, `signal` ASC, time_value ASC 
LIMIT 10000001;
```

If possible, we should switch how we build multiple-signal queries to use UNION instead of OR.

### PR DESCRIPTION
**Pull request by LeonLu2**
closes #763

**Prerequisites**:

- [X] Unless it is a documentation hotfix it should be merged against the `dev` branch
- [X] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [X] Build is successful
- [X] Code is cleaned up and formatted

### Summary

1. Refactored the QueryBuilder to convert covidcast signal OR clauses to UNION
2. Edited test_query.py accordingly to test filters changed in server code

### CODE REVIEW

File: src/server/_query.py
Comment (krivard) [inline src/server/_query.py:421]: Are the linebreaks for readability while debugging? if so, they should be removed before we merge, since they'll probably mess up the logs

File: src/server/_query.py
Comment (krivard) [inline src/server/_query.py:420]: is `i` strictly necessary here? why not use something like

```
for signal_clause in self.signal_array[1:]:
```

?

File: src/server/_query.py
Comment (krivard) [reply src/server/_query.py:421]: how does `GROUP BY` interact with `UNION ALL`? if we're grouping by something other than source/signal, will the collected results be grouped correctly or does `UNION` impose an implicit grouping by the `signal_array` column(s)?

File: src/server/_query.py
Comment (krivard) [inline src/server/_query.py:553]: alas, we have more columns in epidata that use reserved words than just `signal` -- is there a downside to escaping all the columns? e.g.
```
args_order = [f"`{k}` ASC" for k in args]
```

File: src/server/_query.py
Comment (krivard) [reply src/server/_query.py:420]: or even something like 

```
"UNION ALL".join(
    f"SELECT {self.fields_clause} FROM {self.table} {index} {self.subquery} {where} AND {signal_clause} {group_by}"
    for signal_clause in self.signal_array
)
```

File: src/server/_query.py
Comment (krivard) [inline src/server/_query.py:558]: looks like we'll end up with a mix of columns in ORDER BY, some without a table prefix, and some with, depending on whether they come from *args or **kwargs when `set_order` is called. Is that intended?

File: tests/server/test_query.py
Comment (krivard) [inline tests/server/test_query.py:212]: 💯

**Review by krivard — COMMENTED** [review]
this is your first major change to the code, so I was pickier than I would otherwise be.

a few logic confirmations which need to be addressed before approval, but mostly style suggestions

also: are the `filter_X` equivalents to the new `alternative_filter_X` functions being used anymore? If not we can just drop them.

File: src/server/_query.py
Comment (LeonLu2) [reply src/server/_query.py:552]: I changed here to all without the table prefix.

File: src/server/_query.py
Comment (LeonLu2) [reply src/server/_query.py]: Edited √

File: src/server/_query.py
Comment (LeonLu2) [reply src/server/_query.py]: Yeah, I used the second one

File: src/server/_query.py
Comment (LeonLu2) [reply src/server/_query.py]: Linebreaks removed. Group by and Order by added at the end of the query.

**Review by krivard — APPROVED** [review]
I am reasonably satisfied, but I'm looping in Dmitry and George on this since it conflicts/interacts with a possible solution to a problem they're solving in an upcoming feature release for just-in-time processing.

**Comment by melange396** [comment]
The issue this was intended to address (#763) is no longer relevant, so i am closing this PR.

**Pull request closed without merging**
